nuxeo-platform-video-core/src/test/java/org/nuxeo/ecm/platform/video/tools/TestVideoToolsOperations.java

```diff
@@ -0,0 +1,253 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Ricardo Dias
+ */
+package org.nuxeo.ecm.platform.video.tools;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.common.utils.FileUtils;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationChain;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.core.util.BlobList;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.impl.DocumentModelListImpl;
+import org.nuxeo.ecm.core.api.impl.blob.FileBlob;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.platform.video.VideoHelper;
+import org.nuxeo.ecm.platform.video.VideoInfo;
+import org.nuxeo.ecm.platform.video.tools.operations.AddWatermarkToVideo;
+import org.nuxeo.ecm.platform.video.tools.operations.ConcatVideos;
+import org.nuxeo.ecm.platform.video.tools.operations.ExtractClosedCaptionsFromVideo;
+import org.nuxeo.ecm.platform.video.tools.operations.SliceVideo;
+import org.nuxeo.ecm.platform.video.tools.operations.SliceVideoInParts;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.google.inject.Inject;
+import org.nuxeo.runtime.test.runner.LocalDeploy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @since 8.4
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class })
+@RepositoryConfig(cleanup = Granularity.METHOD)
+@Deploy({ "org.nuxeo.ecm.automation.core", "org.nuxeo.ecm.platform.video.convert",
+        "org.nuxeo.ecm.platform.picture.core" })
+@LocalDeploy({ "org.nuxeo.ecm.platform.video.core:OSGI-INF/core-types-contrib.xml",
+        "org.nuxeo.ecm.platform.video.core:OSGI-INF/video-tools-commandlines-contrib.xml",
+        "org.nuxeo.ecm.platform.video.core:OSGI-INF/video-tools-operations-contrib.xml",
+        "org.nuxeo.ecm.platform.video.core:OSGI-INF/video-tools-service.xml" })
+public class TestVideoToolsOperations extends BaseVideoToolsTest {
+
+    public static final String WATERMARK_PICTURE = "test-data/logo.jpeg";
+
+    protected DocumentModel folder;
+
+    @Inject
+    CoreSession coreSession;
```

Comment:
Use an explicit `protected` (for instance), don't use package-private visibility.